src/core/expr/head_reduce_unary.cc

```diff
@@ -183,248 +183,6 @@ static Column compute_gfirstlast(Column&& arg, const Groupby&) {
-                           std::move(arg), gby
-                         ));
-    default: throw _error("prod", arg.stype());
-  }
-}
-
-
-
-//------------------------------------------------------------------------------
-// sum(A)
-//------------------------------------------------------------------------------
-
-template <typename T, typename U>
-bool sum_reducer(const Column& col, size_t i0, size_t i1, U* out) {
-  U sum = 0;
-  for (size_t i = i0; i < i1; ++i) {
-    T value;
-    bool isvalid = col.get_element(i, &value);
-    if (isvalid) {
-      sum += static_cast<U>(value);
-    }
-  }
-  *out = sum;
-  return true;  // *out is not NA
-}
-
-
-
-template <typename T, typename U>
-static Column _sum(Column&& arg, const Groupby& gby) {
-  return Column(
-          new Latent_ColumnImpl(
-            new Reduced_ColumnImpl<T, U>(
-                 stype_from<U>, std::move(arg), gby, sum_reducer<T, U>
-            )));
-}
-
-static Column compute_sum(Column&& arg, const Groupby& gby) {
-  switch (arg.stype()) {
-    case SType::VOID:    return Const_ColumnImpl::make_int_column(gby.size(), 0, SType::INT64);
-    case SType::BOOL:
-    case SType::INT8:    return _sum<int8_t, int64_t>(std::move(arg), gby);
-    case SType::INT16:   return _sum<int16_t, int64_t>(std::move(arg), gby);
-    case SType::INT32:   return _sum<int32_t, int64_t>(std::move(arg), gby);
-    case SType::INT64:   return _sum<int64_t, int64_t>(std::move(arg), gby);
-    case SType::FLOAT32: return _sum<float, float>(std::move(arg), gby);
-    case SType::FLOAT64: return _sum<double, double>(std::move(arg), gby);
-    default: throw _error("sum", arg.stype());
-  }
-}
-
-
-
-
-//------------------------------------------------------------------------------
-// sum(A:grouped)
-//------------------------------------------------------------------------------
-
-// T - type of input elements in the `arg` column
-// U - type of output elements
-template <typename T, typename U>
-class SumGrouped_ColumnImpl : public Virtual_ColumnImpl {
-  private:
-    Column arg;
-    Groupby groupby;
-
-  public:
-    SumGrouped_ColumnImpl(Column&& col, const Groupby& grpby)
-      : Virtual_ColumnImpl(grpby.size(), stype_from<U>),
-        arg(std::move(col)),
-        groupby(grpby)
-        {}
-
-    ColumnImpl* clone() const override {
-      return new SumGrouped_ColumnImpl<T, U>(Column(arg), groupby);
-    }
-
-    size_t n_children() const noexcept override {
-      return 1;
-    }
-
-    const Column& child(size_t i) const override {
-      xassert(i == 0);  (void)i;
-      return arg;
-    }
-
-    bool get_element(size_t i, U* out) const override {
-      T value;
-      size_t i0, i1;
-      groupby.get_group(i, &i0, &i1);
-      bool isvalid = arg.get_element(i, &value);
-      *out = isvalid? static_cast<U>(i1 - i0) * static_cast<U>(value)
-                    : U(0);
-      return true; // *out is never an NA
-    }
```

Comment:
Look at this class and this method. This is how the sum could be calculated for a grouped column. You don't need to go through all the elements, because they're all equal, so you just multiply the size of group by only one value.